packages/lit-analyzer/src/analyze/util/type/is-assignable-binding-under-security-system.ts

```diff
@@ -0,0 +1,109 @@
+import { HtmlNodeAttr } from "../../types/html-node/html-node-attr-types";
+import { SimpleType, toTypeString, SimpleTypeKind } from "ts-simple-type";
+import { LitAnalyzerRequest } from "../../lit-analyzer-context";
+import { LitHtmlDiagnostic, LitHtmlDiagnosticKind } from "../../types/lit-diagnostic";
+import { LitAnalyzerRuleName } from "../../lit-analyzer-config";
+
+/**
+ * If the user's security policy overrides normal type checking for this
+ * attribute binding, returns a (possibly empty) array of diagnostics.
+ *
+ * If the security policy does not apply to this binding, then
+ */
+export function isAssignableBindingUnderSecuritySystem(
+	htmlAttr: HtmlNodeAttr,
+	{ typeA, typeB }: { typeA: SimpleType; typeB: SimpleType },
+	request: LitAnalyzerRequest,
+	source: LitAnalyzerRuleName
+): LitHtmlDiagnostic[] | undefined {
+	const securityPolicy = request.config.securitySystem;
+	switch (securityPolicy) {
+		case "off":
+			return undefined; // No security checks apply.
+		case "ClosureSafeTypes":
+			return checkClosureSecurityAssignability(typeB, htmlAttr, request, source);
+		default: {
+			const never: never = securityPolicy;
+			request.logger.error(`Unexpected security policy: ${never}`);
+			return undefined;
+		}
+	}
+}
+
+interface TagNameToSecurityOverrideMap {
+	[tagName: string]: SecurityOverrideMap | undefined;
+}
+
+// A map from attribute/property names to an array of type names.
+// Assignments to the given attribute must match one of the given types.
+interface SecurityOverrideMap {
+	[attributeName: string]: string[] | undefined;
+}
+
+const closureScopedOverrides: TagNameToSecurityOverrideMap = {
+	iframe: {
+		src: ["TrustedResourceUrl"]
+	},
+	a: {
+		href: ["TrustedResourceUrl", "SafeUrl"]
+	},
+	img: {
+		src: ["TrustedResourceUrl", "SafeUrl"]
+	},
+	script: {
+		src: ["TrustedResourceUrl"]
+	}
+};
+const closureGlobalOverrides: SecurityOverrideMap = {
+	style: ["SafeStyle"]
+};
+
+function checkClosureSecurityAssignability(
+	typeB: SimpleType,
+	htmlAttr: HtmlNodeAttr,
+	request: LitAnalyzerRequest,
+	source: LitAnalyzerRuleName
+): LitHtmlDiagnostic[] | undefined {
+	const scopedOverride = closureScopedOverrides[htmlAttr.htmlNode.tagName];
+	const overriddenTypes = (scopedOverride && scopedOverride[htmlAttr.name]) || closureGlobalOverrides[htmlAttr.name];
+	if (overriddenTypes === undefined) {
+		return undefined;
+	}
+	if (!matchesAtLeastOneNominalType(overriddenTypes, typeB)) {
+		const nominalType: SimpleType = {
+			kind: SimpleTypeKind.INTERFACE,
+			members: [],
+			name: "A security type"
+		};
+		return [
+			{
+				kind: LitHtmlDiagnosticKind.INVALID_ATTRIBUTE_EXPRESSION_TYPE,
+				message: `Type '${toTypeString(typeB)}' is not assignable to '${overriddenTypes.join(" | ")}'.`,
```

Comment:
Would it be helpful to add something like "under ${securityPolicy} security policy", so that users get a hint about where to look to understand why they couldn't assign?

Reply:
+1, done